sources/kube_factory.go

```diff
@@ -76,7 +76,18 @@ func getConfigOverrides(uri string, options map[string][]string) (*kubeClientCmd
 	return &kubeConfigOverride, nil
 }
 
-func CreateKubeSources(uri string, options map[string][]string) ([]api.Source, error) {
+func makeKubeConfig(uri string, options map[string][]string) (*kube_client.Config, error) {
+	var kubeConfig *kube_client.Config
+
+	if len(options["inClusterConfig"]) == 1 && options["inClusterConfig"][0] == "true" {
+		kubeConfig, err := kube_client.InClusterConfig()
+		if err != nil {
+			return nil, fmt.Errorf("error creating inClusterConfig: %v", err)
+		}
+		glog.Infof("using in cluster config")
+		return kubeConfig, nil
+	}
+
 	configOverrides, err := getConfigOverrides(uri, options)
 
 	authFile := defaultKubeConfigFile
```

Comment:
Ok. I can do that. Service accounts are on by default but the tls crt is only distributed if the optional `--root-ca-cert` flag is set on the kube-controller-manager. Is that okay?